File: PUSGSProjekat2/ProjekatPUSGS/ProjekatPUSGS/Models/RezervacijaLetova.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace ProjekatPUSGS.Models
{
    public class RezervacijaLetova
    {
        [Key]
        public int Id { get; set; }

        public int IdAvioKompanije { get; set; }

        public int IdLeta { get; set; }

        public double Cena { get; set; }

        public bool Zavrseno { get; set; }

        public string IdKlijenta { get; set; }

        public DateTime PocetniDatum { get; set; }

        public DateTime KrajnjiDatum { get; set; }

        public int OcenaZaKomapaniju { get; set; }
        public int OcenaZaLet { get; set; }

        [NotMapped]
        public List<DateTime> ZauzetiDatumi { get; set; }

        public string DatumiZauzeti { get; set; }

        [NotMapped]
        public Letovi Destinacija { get; set; }


        public void PretvaranjeiUJson()
        {
            string output = JsonConvert.SerializeObject(ZauzetiDatumi);
            DatumiZauzeti = output;
        }

        public void PretvaranjeUListu()
        {
            if (DatumiZauzeti != null)
            {
                List<DateTime> deserializedProduct = JsonConvert.DeserializeObject<List<DateTime>>(DatumiZauzeti);
                ZauzetiDatumi = deserializedProduct;
            }
        }
    }
}

File: PUSGSProjekat/src/app/entities/rezervacija-letova/rezervacija-letova.ts
import {NumberValueAccessor} from '@angular/forms';
import {Letovi} from '../letovi/letovi';

export class RezervacijaLetova{
    id:number;
    idAvioKompanije:number;
    idLeta:number;
    idKlijenta:string;
    cena:number;
    pocetniDatum:string;
    krajnjiDatum:string;
    destinacija:Letovi;
    zavrseno:boolean;

    constructor(id:number, idAvioKompanije:number, idLeta:number, idKlijenta:string, cena:number, pocetniDatum:string, krajnjiDatum:string, destinacija:Letovi) {
        this.id=id;
        this.idAvioKompanije=idAvioKompanije;
        this.idLeta=idLeta;
        this.idKlijenta=idKlijenta;
        this.cena=cena;
        this.pocetniDatum=pocetniDatum;
        this.krajnjiDatum=krajnjiDatum;
        this.destinacija=destinacija;
    }


}